gobblin-core/src/main/java/gobblin/converter/EnvelopeSchemaConverter.java

```diff
@@ -0,0 +1,86 @@
+package gobblin.converter;
+
+import com.google.common.base.Optional;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.WorkUnitState;
+import gobblin.converter.filter.AvroProjectionConverter;
+import gobblin.converter.filter.AvroSchemaFieldRemover;
+import gobblin.metrics.kafka.KafkaAvroSchemaRegistry;
+import gobblin.metrics.kafka.SchemaRegistryException;
+import gobblin.util.AvroUtils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import javax.xml.bind.DatatypeConverter;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+
+/**
+ * A converter for opening GoldenGate envelope schema.
+ */
+public class EnvelopeSchemaConverter extends AvroToAvroConverterBase {
+
+  public static final String PAYLOAD_SCHEMA_ID = "envelope.schema.id.field";
```

Comment:
1. All properties should have namespace to prevent name clashes.
2. Dots.are.not.substitutes.for.spaces. Dots should be used as scope separators. E.g. envelope.schemaIdField, envelope.payloadField.